### PR TITLE
feat(CurriculumCard): updated footer text variant

### PR DESCRIPTION
**Pull request by amabarr**
### Overview

Teams+ users want to see the copy "In Progress..." instead of "Enrolled..." for their own organization's content. 

So, I made an optional variable called `footerTextVariant` that either takes in `inProgress` or `enrolled`, and displays the corresponding text.  

### PR Checklist

- [x] Related to designs: [figma](https://www.figma.com/file/YR6abM4OmOq0Q8YrifzOpn/Self-Service-Course-%26-Path-Release?node-id=492%3A5861)
- [x] Related to JIRA ticket: [CONPLAT-1121](https://codecademy.atlassian.net/secure/RapidBoard.jspa?rapidView=156&modal=detail&selectedIssue=CONPLAT-1121)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
LGTM! Unit test please? 🕺

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.90-alpha.a79975.0<br/>@codecademy/gamut-labs@23.0.1-alpha.a79975.0<br/>@codecademy/styleguide@46.0.1-alpha.a79975.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614b5884fb9b01013c74d1a7--gamut-preview.netlify.app>https://614b5884fb9b01013c74d1a7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614b5884fb9b01013c74d1a7>Deploy Logs</a></p>

File: packages/gamut-labs/src/CurriculumCard/Footer/__tests__/Footer-test.tsx [hunk 1]
```diff
@@ -42,4 +42,17 @@ describe('CurriculumCard Footer', () => {
     const { view } = renderView({ beta: true, progressState: 'completed' });
     expect(view.queryByText('BETA')).toBeFalsy();
   });
+
+  it('displays enrolled footer variant by default', () => {
+    const { view } = renderView({ progressState: 'inProgress' });
+    expect(view.queryByText('Enrolled...')).toBeTruthy();
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/CurriculumCard/Footer/__tests__/Footer-test.tsx:48]: ⛳ ✨ 

```suggestion
    view.getByText('Enrolled...'));
```

**Review by JoshuaKGoldberg — APPROVED** [review]
🌮

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.91-alpha.75f8e9.0<br/>@codecademy/gamut-labs@23.0.2-alpha.75f8e9.0<br/>@codecademy/styleguide@46.1.1-alpha.75f8e9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://614e02e2b91f890a0467665e--gamut-preview.netlify.app>https://614e02e2b91f890a0467665e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/614e02e2b91f890a0467665e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.93-alpha.fe602f.0<br/>@codecademy/gamut-labs@23.2.1-alpha.fe602f.0<br/>@codecademy/styleguide@46.3.1-alpha.fe602f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6152059f8b8ed49efcae37a5--gamut-preview.netlify.app>https://6152059f8b8ed49efcae37a5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6152059f8b8ed49efcae37a5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.93-alpha.94035c.0<br/>@codecademy/gamut-labs@23.2.1-alpha.94035c.0<br/>@codecademy/styleguide@46.3.2-alpha.94035c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61522b1e79d2cf2f6ead89f6--gamut-preview.netlify.app>https://61522b1e79d2cf2f6ead89f6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61522b1e79d2cf2f6ead89f6>Deploy Logs</a></p>

**Pull request merged**